### DISCUSSION TITLE
CME Initial Profiling

### DISCUSSION THREAD

**Discussion by wilfchun** [coastalme/coastalme#9] [General]
Hello all,

I have run one of my test cases for 1year at 6hr time steps with no saving of any of the output files.

Looks like calls to GDAL do use a noticeable percent of resources, but there doesn't look to be any low hanging fruit in terms of easy performance gains.

Happy to probe this in more detail if others are interested.
![flamegraph](https://github.com/user-attachments/assets/51326e7c-cb8d-4a5d-ba7e-946d6e8ae723)

**Comment by davefavismortlock** [comment]
Very interesting Wilf! I'm assuming that taller = worse, is that correct? If so, any idea what the red-circled operation is?
![391124737-51326e7c-cb8d-4a5d-ba7e-946d6e8ae723](https://github.com/user-attachments/assets/fe00296d-8a90-430b-8b8d-c5c9db6aa4cd)

**Reply by wilfchun** [reply]
Not exactly, this is a flame chart which I guess is essentially a unravelled pie chart. The bottom (grey) row is the total run time, then as we go up we go deeper into routines and subroutines.
The wider a bar is the larger percentage of the total runtime that specific command took.
https://www.polarsignals.com/blog/posts/2023/03/28/how-to-read-icicle-and-flame-graphs
